### DISCUSSION TITLE
Configuration files for a bolt base swarm

### DISCUSSION THREAD

**Discussion by ghost** [USC-ACTLab/crazyswarm#775] [Q&A]
Hello,

A friend and I are working on a drone swarm project based on the Crazyflie Bolt platform (same drone built multiple time). And we have some trouble configuring CrazySwarm for our drones, especially the .yaml files.

We have tuned the PID parameters of the Crazyflie firmware using the Crazyflie client, each drone is now stable when controlled by joystick.

As I understood from this page : https://crazyswarm.readthedocs.io/en/latest/configuration.html , I should now define a new type of drone in the "crazyflieTypes.yaml" file and enumerate my drones in the "crazyflies.yaml" file.

But I didn't quite understand how to enter the various parameters for my new type of drone, such as the new PID values (I didn't understand which values corresponded to which line, nor the scale of the values in the "crazyflieTypes.yaml" file).

Thanks in advance

Yours faithfully

**Comment by jpreiss** [comment]
These lines exemplify how to set the params for the Mellinger controller: https://github.com/USC-ACTLab/crazyswarm/blob/master/ros_ws/src/crazyswarm/launch/crazyflieTypes.yaml#L9-L26. If you are using a different controller, you need to change the parameter group and parameter names appropriately. You could use https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/api/params/ as a guide.

**Reply by ghost** [reply]
Thanks for your advices, but I didn't understand how to fill those lines (I am a newbie to crazyswarm and crazyflie). 
How to convert the values form the cfclient parameters on the left to the crazyfliesTypes.yaml file parameters on the right ?
![cfclient_crazyswarm_pid](https://github.com/USC-ACTLab/crazyswarm/assets/136150871/9acac90a-161c-4db4-94b7-bd8b049f4251)

The scale looks to be really different, and i haven't understand what a Mellinger controller is.

Thanks in advance

Your faithfully

**Comment by ghost** [comment]
After some researches, I have understood what the Mellinger controller is.
But I wish to use a PID controller, is-it possible to use a PID controller using Crazyswarm ?

I saw that it is possible to use a PID controller by changing the Crazyflie firmware. But how can I specify to the Crazyswarm API that I am using a PID controller and what is the syntax of a PID controller in the "crazyflieTypes.yaml" file ?

I am really sorry to used a not maintained project, but I can't use Crazyswarm2

Yours faithfully

**Reply by whoenig** [reply]
You can set stabilizer.controller in hover.launch to 1 to use the PID controller.

**Reply by ghost** [reply]
Yes,
It is what I have done, Currently I have managed to tune the rate and attitude PID. I think that they are well tuned because the drone is flying really well with a joystick controller. I am currently tuning the Velocity and position PID (I am struggling a bit but I will managed to get throught it). 
When this would be done I will close the discussion.
